### PR TITLE
Fix MLK related env

### PR DESCRIPTION
**Pull request by qiyuangong**

### CODE REVIEW

File: pyzoo/zoo/util/engine.py [hunk 1]
```diff
@@ -104,7 +104,10 @@ def append_path(env_var_name, path):
     os.environ["KMP_BLOCKTIME"] = "0"
     os.environ["KMP_AFFINITY"] = "granularity=fine,verbose,compact,1,0"
     os.environ["KMP_SETTINGS"] = "1"
-    os.environ["OMP_NUM_THREADS"] = "1"
+    if "ZOO_NUM_MKLTHREADS" in os.environ:
+        os.environ["OMP_NUM_THREADS"] = os.environ["ZOO_NUM_MKLTHREADS"]
+    else:
+        os.environ["OMP_NUM_THREADS"] = "1"
```
Comment (zhichao-li) [inline pyzoo/zoo/util/engine.py]: Maybe we should check the other variable as well including KMP_AFFINITY, KMP_SETTINGS, as the user might want to tune the values and it's not a good idea that always pint the log in verbose mode.

Comment (qiyuangong) [reply pyzoo/zoo/util/engine.py]: Good idea. Will check these parameters in real environment. :)
About verbose model, I think we can add a flag for it.

File: pyzoo/zoo/util/engine.py [hunk 2]
```diff
@@ -101,10 +101,13 @@ def append_path(env_var_name, path):
             if package not in sys.path:
                 sys.path.insert(0, package)
 
-    os.environ["KMP_BLOCKTIME"] = "0"
-    os.environ["KMP_AFFINITY"] = "granularity=fine,verbose,compact,1,0"
+    os.environ["KMP_AFFINITY"] = "granularity=fine,compact,1,0"
     os.environ["KMP_SETTINGS"] = "1"
-    os.environ["OMP_NUM_THREADS"] = "1"
+    if "ZOO_NUM_MKLTHREADS" in os.environ:
+        os.environ["OMP_NUM_THREADS"] = os.environ["ZOO_NUM_MKLTHREADS"]
```
Comment (jason-dai) [inline pyzoo/zoo/util/engine.py]: what about KMP_BLOCKTIME in this case?

Comment (qiyuangong) [reply pyzoo/zoo/util/engine.py]: `KMP_BLOCKTIME=0` will lead to low CPU usage and low performance for OpenVINO.

File: pyzoo/zoo/common/nncontext.py [hunk 3]
```diff
@@ -78,6 +104,12 @@ def get_analytics_zoo_conf():
 
 def init_spark_conf():
     zoo_conf = get_analytics_zoo_conf()
+    # Set bigDL and TF conf
+    zoo_conf["KMP_AFFINITY"] = os.environ["KMP_AFFINITY"]
+    zoo_conf["KMP_SETTINGS"] = os.environ["KMP_SETTINGS"]
+    zoo_conf["OMP_NUM_THREADS"] = os.environ["OMP_NUM_THREADS"]
+    zoo_conf["KMP_BLOCKTIME"] = os.environ["KMP_BLOCKTIME"]
```
Comment (yangw1234) [inline pyzoo/zoo/common/nncontext.py]: should be  zoo_conf["spark.executorEnv.OMP_NUM_THREADS"] = ...?

Comment (qiyuangong) [reply pyzoo/zoo/common/nncontext.py]: Thanks. 
Fixed. :)

File: pyzoo/zoo/common/nncontext.py [hunk 4]
```diff
@@ -76,8 +77,44 @@ def get_analytics_zoo_conf():
...
+    kmp_blocktime = "0"
+
+    # Check env and override if necessary
+    # Currently, focused on ZOO_NUM_MKLTHREADS,
+    # OMP_NUM_THREADS, KMP_BLOCKTIME, KMP_AFFINITY
+    # and KMP_SETTINGS
+    if "OMP_NUM_THREADS" in os.environ:
+        omp_num_threads = os.environ["OMP_NUM_THREADS"]
+    elif "ZOO_NUM_MKLTHREADS" in os.environ:
+        if os.environ["ZOO_NUM_MKLTHREADS"] == "ALL" \
```
Comment (zhichao-li) [inline pyzoo/zoo/common/nncontext.py]: os.environ["ZOO_NUM_MKLTHREADS"].lower() == 'all'

File: pyzoo/zoo/common/nncontext.py [hunk 5]
```diff
@@ -76,8 +77,44 @@ def get_analytics_zoo_conf():
...
+                or os.environ["ZOO_NUM_MKLTHREADS"] == "all":
+            omp_num_threads = multiprocessing.cpu_count()
+        else:
+            omp_num_threads = os.environ["ZOO_NUM_MKLTHREADS"]
+    if "KMP_BLOCKTIME" in os.environ:
```
Comment (zhichao-li) [inline pyzoo/zoo/common/nncontext.py]: Where is the checking for `KMP_AFFINITY` and `KMP_SETTINGS` ?

File: zoo/src/main/scala/com/intel/analytics/zoo/common/NNContext.scala [hunk 6]
```diff
@@ -196,6 +199,43 @@ object NNContext {
...
+  /**
+   * Spark conf with pre-set env
+   * Currently, focus on KMP_AFFINITY, KMP_BLOCKTIME
+   * KMP_SETTINGS, OMP_NUM_THREADS and ZOO_NUM_MKLTHREADS
+   *
+   * @param zooConf SparkConf
+   */
+  private[zoo] def initConf(zooConf: SparkConf) : Unit = {
+    // check env and set spark conf
+    if (!env.contains("KMP_AFFINITY")) {
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/common/NNContext.scala]: What if env indeed contains "KMP_AFFINITY"?  Maybe we can follow the same logic in python: https://github.com/intel-analytics/analytics-zoo/pull/1323/files#diff-6498ec439bfaf5e4e47c0abc1eebd1eaR81

File: pyzoo/zoo/common/nncontext.py (hunk 4 above)
Comment (qiyuangong) [reply pyzoo/zoo/common/nncontext.py]: Fixed. Thanks

File: pyzoo/zoo/common/nncontext.py (hunk 5 above)
Comment (qiyuangong) [reply pyzoo/zoo/common/nncontext.py]: Fixed. Thanks

File: zoo/src/main/scala/com/intel/analytics/zoo/common/NNContext.scala (hunk 6 above)
Comment (qiyuangong) [reply zoo/src/main/scala/com/intel/analytics/zoo/common/NNContext.scala]: Fixed. Thanks :)

**Comment by yangw1234** [comment]
LGTM

**Comment by qiyuangong** [comment]
Jenkins passed http://10.239.47.211:18888/job/ZOO-PR-Validation/2261/ .

**Comment by yangw1234** [comment]
jpyter-with-zoo.sh
pyspark-with-zoo.sh
spark-submit-with-zoo.sh
They all needs to be fixed

**Comment by qiyuangong** [comment]
> jpyter-with-zoo.sh
> pyspark-with-zoo.sh
> spark-submit-with-zoo.sh
> They all needs to be fixed

`jpyter-with-zoo.sh` and 
`pyspark-with-zoo.sh` is missed. Will fix in coming PR.

**Pull request merged**